res/controllers/Numark-N4-scripts.js

```diff
@@ -300,19 +307,29 @@ NumarkN4.MixerTemplate = function () {
     inKey: "mute",
   });
 
-  // NOTE: [Mixer Profile] is not a documented group. Taken from script.crossfaderCurve
-  // BUG: Help on Mixxx forum is needed
   this.changeCrossfaderContour = new components.Button({
     midi: [0x90,0x4B],
+    state: false,
     input: function (channel, control, value, status, group) {
-      if (this.isPress(channel,control, value, status)) {
-        NumarkN4.setCrossfaderSettings({
-          mode: 0, // fast cut (additive)
-          curve: 999.60,
-          calibration: 1.0
+      _.forEach(NumarkN4.crossfaderCallbackConnections, function (callbackObject) {
+        callbackObject.disconnect();
+      });
+      NumarkN4.crossfaderCallbackConnections = [];
+      this.state=this.isPress(channel, control, value, status);
+      if (this.state) {
+        _.forEach(NumarkN4.scratchXFader, function (value, control){
+          engine.setValue("[Mixer Profile]", control, value);
+          NumarkN4.crossfaderCallbackConnections.push(
+            engine.makeConnection("[Mixer Profile]", control, NumarkN4.CrossfaderChangeCallback)
+          );
         });
       } else {
-        NumarkN4.setCrossfaderSettings(NumarkN4.xFaderSettings);
+        _.forEach(NumarkN4.currentCrossfaderParams, function (value, control) {
```

Comment:
The variable name `currentCrossfaderParams` is confusing here because those parameters aren't current when this code is executed. How about renaming it to `storedCrossfaderParams`?